enterprise/server/action_cache_server_proxy/action_cache_server_proxy.go

```diff
@@ -36,19 +50,109 @@ func NewActionCacheServerProxy(env environment.Env) (*ActionCacheServerProxy, er
 	}
 	return &ActionCacheServerProxy{
 		env:         env,
+		localCache:  env.GetCache(),
 		remoteCache: remoteCache,
 	}, nil
 }
 
+func (s *ActionCacheServerProxy) getACKeyForGetActionResultRequest(req *repb.GetActionResultRequest) (*digest.ACResourceName, error) {
+	var sb strings.Builder
+	sb.WriteString(req.GetActionDigest().GetHash())
+	sb.WriteString(*actionCacheSalt)
```

Comment:
This is definitely more of a theoretical concern, but hashing variable length strings without also hashing their lengths could result in unexpected collisions. For example, if `InlineStderr` is disabled but an `e` is added to the salt, the digest wouldn't change.